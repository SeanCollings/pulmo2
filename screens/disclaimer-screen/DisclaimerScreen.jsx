import React, { useContext, useState } from 'react';
import { StyleSheet, View, ScrollView, Text } from 'react-native';

import options from './options';
import optionsEntry from './optionsEntry';
import { useTheme } from '../../hooks/useTheme';
import {
  DISCLAIMER_SHOW,
  DISCLAIMER_DONT_SHOW_AGAIN,
  ProfileContext,
  DISCLAIMER,
} from '../../context/profile-context';
import Checkbox from '../../components/Checkbox';
import CustomButton from '../../components/CustomButton';

export const disclaimerScreenOptions = options;
export const disclaimerScreenOptionsEntry = optionsEntry;

const DisclaimerScreen = ({ route }) => {
  const theme = useTheme();
  const { updateProfileContext } = useContext(ProfileContext);
  const [disclaimerChecked, setDisclaimerChecked] = useState(false);

  const { readOnly } = route.params;

  const opacity = theme.DARK ? 0.87 : 1;
  const textContentStyle = { ...styles.textStyle, color: theme.TEXT, opacity };

  const onAgreeHandler = () => {
    const updatedDisclaimer = {
      [DISCLAIMER_SHOW]: false,
      [DISCLAIMER_DONT_SHOW_AGAIN]: disclaimerChecked,
    };
    updateProfileContext(DISCLAIMER, updatedDisclaimer);
  };

  return (
    <View style={{ ...styles.container, backgroundColor: theme.BACKGROUND }}>
      <ScrollView
        style={styles.scrollviewContainer}
        contentContainerStyle={{ alignSelf: 'center' }}
      >
        <View style={styles.contentContainer}>
          <Text style={textContentStyle}>
            The use of this application is not intended to be a substitute for
            professional medical advice, diagnosis, or treatment. Always seek
            the advice of your physician or other qualified health provider with
            any questions or queries you may have.
          </Text>
          <Text style={textContentStyle} />
          <Text style={textContentStyle}>
            Rinse your Luft&trade; thoroughly with warm water and a mild
            detergent after each use. Be sure to flush both ends of the device
            and any attachments that you may have used with clear water. Shake
            out any excess water and allow to air dry. DO NOT BOIL or HEAT.
            After allowing the device to dry completely, store the Luft&trade;
            for your next session.
          </Text>
          <Text style={textContentStyle} />
          <Text style={textContentStyle}>Note:</Text>
          <Text style={textContentStyle}>
            For asthmatics, do not exhale using the Luft&trade; in the first
            week, rather breathe out normally. During the second week, exhale
            using the Luft&trade;.
          </Text>
          <Text style={textContentStyle} />
          {!readOnly && (
            <View style={styles.selectionContainer}>
              <View style={styles.checkboxContainer}>
                <Checkbox
                  state={disclaimerChecked}
                  onChange={() => setDisclaimerChecked((curr) => !curr)}
                  label="Don't show me again"
                />
              </View>
              <View style={styles.buttonContainer}>
                <CustomButton
                  title="Agree"
                  onPress={onAgreeHandler}
                  style={{ width: 100 }}
                />
              </View>
            </View>
          )}
        </View>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
  },
  scrollviewContainer: {
    width: '100%',
  },
  contentContainer: { alignItems: 'center', width: '80%', paddingTop: 20 },
  upperContainer: {
    flex: 1,
    width: '100%',
    alignItems: 'center',
  },
  textStyle: {
    fontFamily: 'tit-regular',
    fontSize: 16,
    textAlign: 'center',
  },
  selectionContainer: { alignItems: 'center' },
  checkboxContainer: {
    paddingTop: 10,
  },
  buttonContainer: {
    paddingTop: 10,
    paddingBottom: 30,
  },
});

export default DisclaimerScreen;
