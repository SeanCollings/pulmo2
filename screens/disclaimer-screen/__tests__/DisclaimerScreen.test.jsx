import React from 'react';
import { act, create } from 'react-test-renderer';
import { cleanup } from 'react-native-testing-library';
import DisclaimerScreen, {
  disclaimerScreenOptions,
  disclaimerScreenOptionsEntry,
} from '../DisclaimerScreen';
import ProfileContextProvider from '../../../context/profile-context';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

describe('DisclaimerScreen - unit test', () => {
  let tree;
  const routeTrue = { params: { readOnly: true } };
  const routeFalse = { params: { readOnly: false } };

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test('should render readOnly true', async () => {
    await act(async () => {
      tree = create(
        <ProfileContextProvider>
          <DisclaimerScreen route={routeTrue} />
        </ProfileContextProvider>
      );
    });

    expect(tree).toMatchSnapshot();
  });

  test('should render readOnly false', async () => {
    await act(async () => {
      tree = create(
        <ProfileContextProvider>
          <DisclaimerScreen route={routeFalse} />
        </ProfileContextProvider>
      );
    });

    expect(tree).toMatchSnapshot();
  });

  describe('disclaimerScreenOptions', () => {
    test('options', () => {
      const options = disclaimerScreenOptions();
      expect(options).toMatchSnapshot();
    });

    test('headerTitle', () => {
      const headerTitle = disclaimerScreenOptions().headerTitle({
        tintColor: 'white',
      });
      expect(headerTitle).toMatchSnapshot();
    });
  });

  describe('disclaimerScreenOptionsEntry', () => {
    test('optionsEntry', () => {
      const optionsEntry = disclaimerScreenOptionsEntry();
      expect(optionsEntry).toMatchSnapshot();
    });

    test('headerTitle', () => {
      const headerTitle = disclaimerScreenOptionsEntry().headerTitle({
        tintColor: 'white',
      });
      expect(headerTitle).toMatchSnapshot();
    });
  });
});
