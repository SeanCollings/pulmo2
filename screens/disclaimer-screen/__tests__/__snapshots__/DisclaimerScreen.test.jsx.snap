// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisclaimerScreen - unit test disclaimerScreenOptions headerTitle 1`] = `
<Text
  style={
    Object {
      "color": "white",
      "fontSize": 20,
    }
  }
>
  Disclaimer
</Text>
`;

exports[`DisclaimerScreen - unit test disclaimerScreenOptions options 1`] = `
Object {
  "cardStyleInterpolator": [Function],
  "headerTitle": [Function],
}
`;

exports[`DisclaimerScreen - unit test disclaimerScreenOptionsEntry headerTitle 1`] = `
<Text
  style={
    Object {
      "alignSelf": "center",
      "color": "white",
      "fontSize": 20,
    }
  }
>
  Disclaimer
</Text>
`;

exports[`DisclaimerScreen - unit test disclaimerScreenOptionsEntry optionsEntry 1`] = `
Object {
  "headerTitle": [Function],
}
`;

exports[`DisclaimerScreen - unit test should render readOnly false 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignSelf": "center",
      }
    }
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "paddingTop": 20,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#3e4b57",
              "fontFamily": "tit-regular",
              "fontSize": 16,
              "opacity": 1,
              "textAlign": "center",
            }
          }
        >
          The use of this application is not intended to be a substitute for professional medical advice, diagnosis, or treatment. Always seek the advice of your physician or other qualified health provider with any questions or queries you may have.
        </Text>
        <Text
          style={
            Object {
              "color": "#3e4b57",
              "fontFamily": "tit-regular",
              "fontSize": 16,
              "opacity": 1,
              "textAlign": "center",
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#3e4b57",
              "fontFamily": "tit-regular",
              "fontSize": 16,
              "opacity": 1,
              "textAlign": "center",
            }
          }
        >
          Rinse your Luft™ thoroughly with warm water and a mild detergent after each use. Be sure to flush both ends of the device and any attachments that you may have used with clear water. Shake out any excess water and allow to air dry. DO NOT BOIL or HEAT. After allowing the device to dry completely, store the Luft™ for your next session.
        </Text>
        <Text
          style={
            Object {
              "color": "#3e4b57",
              "fontFamily": "tit-regular",
              "fontSize": 16,
              "opacity": 1,
              "textAlign": "center",
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#3e4b57",
              "fontFamily": "tit-regular",
              "fontSize": 16,
              "opacity": 1,
              "textAlign": "center",
            }
          }
        >
          Note:
        </Text>
        <Text
          style={
            Object {
              "color": "#3e4b57",
              "fontFamily": "tit-regular",
              "fontSize": 16,
              "opacity": 1,
              "textAlign": "center",
            }
          }
        >
          For asthmatics, do not exhale using the Luft™ in the first week, rather breathe out normally. During the second week, exhale using the Luft™.
        </Text>
        <Text
          style={
            Object {
              "color": "#3e4b57",
              "fontFamily": "tit-regular",
              "fontSize": 16,
              "opacity": 1,
              "textAlign": "center",
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <View
            style={
              Object {
                "paddingTop": 10,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="checkboxButton"
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#CCC",
                          "fontSize": 28,
                        },
                        Object {
                          "opacity": 1,
                        },
                        Object {
                          "fontFamily": "material-community",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                  <View
                    style={
                      Object {
                        "position": "absolute",
                        "transform": Array [
                          Object {
                            "scale": 0.01,
                          },
                        ],
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#3e4b57",
                            "fontSize": 16,
                          },
                          Object {
                            "opacity": 1,
                          },
                          Object {
                            "fontFamily": "material-community",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      ﹮
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "justifyContent": "center",
                      "paddingLeft": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#3e4b57",
                        "fontFamily": "tit-light",
                        "fontSize": 16,
                        "opacity": 1,
                        "paddingBottom": 2,
                      }
                    }
                  >
                    Don't show me again
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "paddingBottom": 30,
                "paddingTop": 10,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "#CCC",
                    "borderRadius": 8,
                    "borderWidth": 1,
                    "justifyContent": "center",
                    "paddingHorizontal": 14,
                    "paddingVertical": 8,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.25,
                    "shadowRadius": 2.84,
                    "width": 100,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#3e4b57",
                      "fontFamily": "tit-regular",
                      "fontSize": 14,
                      "opacity": 1,
                      "textAlign": "center",
                      "textTransform": "uppercase",
                    }
                  }
                >
                  Agree
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`DisclaimerScreen - unit test should render readOnly true 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignSelf": "center",
      }
    }
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "paddingTop": 20,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#3e4b57",
              "fontFamily": "tit-regular",
              "fontSize": 16,
              "opacity": 1,
              "textAlign": "center",
            }
          }
        >
          The use of this application is not intended to be a substitute for professional medical advice, diagnosis, or treatment. Always seek the advice of your physician or other qualified health provider with any questions or queries you may have.
        </Text>
        <Text
          style={
            Object {
              "color": "#3e4b57",
              "fontFamily": "tit-regular",
              "fontSize": 16,
              "opacity": 1,
              "textAlign": "center",
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#3e4b57",
              "fontFamily": "tit-regular",
              "fontSize": 16,
              "opacity": 1,
              "textAlign": "center",
            }
          }
        >
          Rinse your Luft™ thoroughly with warm water and a mild detergent after each use. Be sure to flush both ends of the device and any attachments that you may have used with clear water. Shake out any excess water and allow to air dry. DO NOT BOIL or HEAT. After allowing the device to dry completely, store the Luft™ for your next session.
        </Text>
        <Text
          style={
            Object {
              "color": "#3e4b57",
              "fontFamily": "tit-regular",
              "fontSize": 16,
              "opacity": 1,
              "textAlign": "center",
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#3e4b57",
              "fontFamily": "tit-regular",
              "fontSize": 16,
              "opacity": 1,
              "textAlign": "center",
            }
          }
        >
          Note:
        </Text>
        <Text
          style={
            Object {
              "color": "#3e4b57",
              "fontFamily": "tit-regular",
              "fontSize": 16,
              "opacity": 1,
              "textAlign": "center",
            }
          }
        >
          For asthmatics, do not exhale using the Luft™ in the first week, rather breathe out normally. During the second week, exhale using the Luft™.
        </Text>
        <Text
          style={
            Object {
              "color": "#3e4b57",
              "fontFamily": "tit-regular",
              "fontSize": 16,
              "opacity": 1,
              "textAlign": "center",
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
